hw/mcu/nordic/nrf52xxx/src/nrf52_clock.c

```diff
@@ -44,7 +44,18 @@ nrf52_clock_hfxo_request(void)
     __HAL_DISABLE_INTERRUPTS(ctx);
     assert(nrf52_clock_hfxo_refcnt < 0xff);
     if (nrf52_clock_hfxo_refcnt == 0) {
-        NRF_CLOCK->TASKS_HFCLKSTART = 1;
+        /* Check the current STATE and SRC of HFCLK */
+        if ((NRF_CLOCK->HFCLKSTAT &
+             (CLOCK_HFCLKSTAT_SRC_Msk | CLOCK_HFCLKSTAT_STATE_Msk)) !=
+            (CLOCK_HFCLKSTAT_SRC_Msk | CLOCK_HFCLKSTAT_STATE_Msk)) {
```

Comment:
Just out of  curiosity... What is this condition supposed to test?

Reply:
I was speaking to @ncasaril about this. The specification shows two bits in that register. SRC is bit 0 and is 0 for HFINT and 1 for HFXO. The STATE bit is bit 16 and is 0 if HFCLK not running and 1 if running. So it seemed best to check both bits being 1. It might be possible to just check SRC and if 1 all is good. Looking at one of the versions of the SDK from nordic it does appear the check both bits.

Reply:
Okay, then this is not very obvious... I would suggest comparing to `CLOCK_HFCLKSTAT_SRC_Xtal` and `CLOCK_HFCLKSTAT_STATE_Running` defines, something like:
```
        if ((NRF_CLOCK->HFCLKSTAT &
             (CLOCK_HFCLKSTAT_SRC_Msk | CLOCK_HFCLKSTAT_STATE_Msk)) !=
            (CLOCK_HFCLKSTAT_SRC_Xtal << CLOCK_HFCLKSTAT_SRC_Pos | 
             CLOCK_HFCLKSTAT_STATE_Running << CLOCK_HFCLKSTAT_STATE_Pos)) {
```

Reply:
I agree that this is clearer. The reason we used the other if-statment was that this was the original statement used and I wanted to keep the changes to a minimum. I'll change so it's clearer.